core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -209,15 +202,11 @@ protected DiffRepository getDiffRepos() {
             + "where case when (sal = 1000) then\n"
             + "(case when sal = 1000 then null else 1 end is null) else\n"
             + "(case when sal = 2000 then null else 1 end is null) end is true";
-    sql(sql).withPre(preProgram)
-        .with(hepPlanner)
+    sql(sql).with(hepPlanner)
```

Comment:
Can we also have `withRuleClass` or `withPreRuleClass` ?

Reply:
I don't think so. I wanted to just cover the common case (simple programs with a few rules). For complex cases, people should continue to explicitly build programs.